src/trigger.ts

```diff
@@ -15,422 +15,454 @@ export const ALL_LAYERS = 255
+				targetEngine.removeEntity(shapes.pop()!)
+			}
+			while (shapes.length < areaCount) {
+				shapes.push(targetEngine.addEntity())
+			}
+
+			const worldPosition = getWorldPosition(entity)
+			const worldRotation = getWorldRotation(entity)
+
+			for (let i = 0; i < areaCount; ++i) {
+				const shapeSpec = trigger.areas[i]
+				const shape = shapes[i]
+
+				let scale
+				if (shapeSpec.$case == 'box') {
+					scale = shapeSpec.value.scale
+					MeshRenderer.setBox(shape)
+				} else {
+					const radius = shapeSpec.value.radius
+					scale = { x: radius, y: radius, z: radius }
+					MeshRenderer.setSphere(shape)
+				}
+
+				Transform.createOrReplace(shape, {
+					position: Vector3.add(worldPosition, Vector3.rotate(shapeSpec.value.position, worldRotation)),
+					scale: scale
+				})
+
+				const color = trigger.active ? trigger.debugColor : Color3.Black()
+				Material.setPbrMaterial(shape, { albedoColor: Color4.fromInts(255 * color.r, 255 * color.g, 255 * color.b, 75) })
+			}
+		}
+	}
+
+	function areTriggersIntersecting(
+		shapeWorldPos0: Array<Vector3>,
+		t0: DeepReadonly<TriggerType>,
+		shapeWorldPos1: Array<Vector3>,
+		t1: DeepReadonly<TriggerType>
+	): boolean {
+		for (let i = 0; i < t0.areas.length; ++i) {
+			const t0World = shapeWorldPos0[i]
+			const t0Area = t0.areas[i]
+
+			if (t0Area.$case == 'box') {
+				const t0Box = t0Area.value
+				const t0Min = Vector3.subtract(t0World, Vector3.scale(t0Box.scale, 0.5))
+				const t0Max = Vector3.add(t0Min, t0Box.scale)
+
+				for (let j = 0; j < t1.areas.length; ++j) {
+					const t1World = shapeWorldPos1[j]
+					const t1Area = t1.areas[j]
+
+					if (t1Area.$case == 'box') {
+						const t1Box = t1Area.value
+						const t1Min = Vector3.subtract(t1World, Vector3.scale(t1Box.scale, 0.5))
+						const t1Max = Vector3.add(t1Min, t1Box.scale)
+
+						if (areAABBIntersecting(t0Min, t0Max, t1Min, t1Max))
+							return true
+					} else {
+						if (areAABBSphereIntersecting(t0Min, t0Max, t1World, t1Area.value.radius))
+							return true
+					}
+				}
+			} else {
+				const t0Radius = t0Area.value.radius
+
+				for (let j = 0; j < t1.areas.length; ++j) {
+					const t1World = shapeWorldPos1[j]
+					const t1Area = t1.areas[j]
+
+					if (t1Area.$case == 'box') {
+						const t1Box = t1Area.value
+						const t1Min = Vector3.subtract(t1World, Vector3.scale(t1Box.scale, 0.5))
+						const t1Max = Vector3.add(t1Min, t1Box.scale)
+
+						if (areAABBSphereIntersecting(t1Min, t1Max, t0World, t0Radius))
+							return true
+					} else {
+						if (areSpheresIntersecting(t0World, t0Radius, t1World, t1Area.value.radius))
+							return true
+					}
+				}
+			}
+		}
+
+		return false
+	}
+
+	function computeCollisions(entity: Entity, shapeWorldPos: Map<Entity, Array<Vector3>>) {
+		let collisions: Set<Entity> = new Set()
+		const trigger = Trigger.get(entity)
+
+		if (!trigger.active)
+			return collisions
+
+
+		if (trigger.triggeredByMask == 1) {
```

Comment:
this is backwards breaking if someone registered other triggers as layer 1 to interact with layer 1.  "player multi ball pi machine"   make it 2.0 when release explaining change?